src/main/java/org/example/producer/UserProducer.java

```diff
@@ -0,0 +1,44 @@
+package org.example.producer;
+
+import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig;
+import io.confluent.kafka.serializers.KafkaAvroSerializer;
+import io.confluent.kafka.serializers.subject.RecordNameStrategy;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import java.util.Properties;
+
+public class UserProducer<T> {
+    private final KafkaProducer<String, T> kafkaProducer;
+    private final String topic;
+
+    public UserProducer(String topic, String kafkaServer, String schemaRegistry) {
+        this.topic = topic;
+        Properties properties = new Properties();
+        properties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaServer);
+        properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class);
+        properties.put(AbstractKafkaAvroSerDeConfig.KEY_SUBJECT_NAME_STRATEGY, RecordNameStrategy.class);
+        properties.put(AbstractKafkaAvroSerDeConfig.VALUE_SUBJECT_NAME_STRATEGY, RecordNameStrategy.class);
+        properties.put("schema.registry.url", schemaRegistry);
+        kafkaProducer = new KafkaProducer(properties);
+    }
+
+    public void produce(T t){
+        ProducerRecord<String, T> producerRecord = new ProducerRecord<>(topic, "key", t);
+
+        try {
+            kafkaProducer.send(producerRecord);
```

Comment:
The send method works asynchrounously. Since both producer and consumer reside in the same application you need to wait until message will be published. Otherwise, your consumer will always get 0 records on poll method invocation.
```java
try {
    kafkaProducer.send(producerRecord).get();
} catch (Exception e) {
    throw new RuntimeException(e);
}
```